fastfuels_sdk/exports.py

```diff
@@ -313,6 +326,336 @@ def export_zarr_to_fds(zroot: zarr.hierarchy.Group,
     with open(Path(output_dir, "template.fds"), "w") as fout:
         fout.write(template.substitute(fds_attrs))
 
+def calibrate_duet(zroot: zarr.hierarchy.Group,
```

Comment:
Would you be open to making this into a class to help keep the calibration logic together in the code base? I think that there's enough complexity here that it may be worth considering an API like the following:

```python
class DuetCalibrator:
    def __init__(self, zroot, output_dir, param_dir=None, keep_duet=None, **kwargs):
        self.zroot = zroot
        self.output_dir = Path(output_dir)
        self.param_dir = Path(param_dir) if param_dir else None
        self.keep_duet = keep_duet
        self.kwargs = kwargs
        self.fuel_types = self._get_fuel_types_from_kwargs()
        
    def calibrate(self):
        self._validate_inputs()
        calibration_method = self._get_calibration_method()
        if not calibration_method:
            self._query_landfire_data()
        self._calibrate_data(calibration_method)
        self._write_output()

    def _validate_inputs(self):
        # ... (validation logic here)

    def _get_calibration_method(self):
        # ... (determine calibration method logic)

    def _query_landfire_data(self):
        # ... (LandFire query logic)

    def _calibrate_data(self, method):
        # ... (calibration logic based on the method)

    def _write_output(self):
        # ... (write to output logic)
```